projects/odf_transform/odf_transform/odf_parser/odf_parser.py

```diff
@@ -199,34 +190,103 @@ def _find_previous_key(key_list, present_key):
+            flag_name_with_no_prefix = re.sub(r"quality\sflag.*:\s*|quality flag of ", '',
+                                              att['original_NAME'], 1, re.IGNORECASE)
+            # Flag name do not match either variable name or code, give a warning.
+            if related_variable and \
+                    flag_name_with_no_prefix not in metadata[related_variable].get('original_NAME') and \
+                    flag_name_with_no_prefix not in metadata[related_variable].get('original_CODE'):
+                warnings.warn(
+                    '{0}[{4}] flag was matched to referring to {1} but odf variable name[{2}] or code[{3}] do not match'
+                        .format(var,
+                                related_variable,
+                                metadata[related_variable].get('original_NAME'),
+                                metadata[related_variable].get('original_CODE'),
+                                att['original_NAME']),
+                    UserWarning)
+
+        # Standardize Flag variable attributes, related variable and add convention attributes
+        if related_variable:
+            # Add long name attribute which is generally QUALITY_FLAG: [variable it affects]
+            if 'name' in metadata[related_variable]:
+                att['long_name'] = flag_long_name_prefix + metadata[related_variable]['name']
+            else:
+                att['long_name'] = flag_long_name_prefix + related_variable
+
+            # Add ancillary_variables attribute
+            if 'ancillary_variables' not in metadata[related_variable]:
+                metadata[related_variable]['ancillary_variables'] = var
+            elif 'ancillary_variables' in metadata[related_variable] and type(metadata[related_variable]) is str:
+                metadata[related_variable]['ancillary_variables'] += ',{0}'.format(var)
+            else:
+                warnings.warn('unknown ancillary flag format attribute', UserWarning)
+
+        # Add flag convention attributes if available within config file
+        if flag_convention:
+            if var in flag_convention:
+                att.update(flag_convention[var])
+            elif 'default' in flag_convention:
+                att.update(flag_convention['default'])
+
+        # TODO rename QQQQ_XX flag variables to Q[related_variables] so that ERDDAP can easily amalgamate them!
+
+    return metadata
+
+
+def get_vocabulary_attributes(metadata,
+                              organizations=None,
+                              vocabulary=None,
+                              vocabulary_attribute_list=None
+                              ):
+    """
+    This method is use to retrieve from an ODF file each variable code and corresponding related
+    vocabularies associated to the organization and variable name.
+    Flag columns are also reviewed and matched to the appropriate variable.
+    """
+    # Make sure organization is a list
+    if type(organizations) is str:
```

Comment:
is there a case where organizations would not be a string?

Reply:
We'll it could always be a list of strings actually. There's a higher change it will be always a list

Reply:
This if was removed and we're now assuming that the `organisationVocabulary` input within the configuration file will always be a list.